### ISSUE TITLE
React support

### ISSUE THREAD

**Issue by jlarmstrongiv** [NiklasPor/prettier-plugin-organize-attributes#10]

**Comment by dycw** [comment]
Absolutely would love this!

**Comment by aronmal** [comment]
I have not thought about it, but I would totally try this plugin, if it would support react.

**Comment by bassem97** [comment]
Insist on introducing this feature

**Comment by NiklasPor** [comment]
Hey all, this formatter is based on a html parser which reorders the attributes. I'm also using TSX a lot and would love react support, but I think this would require a lot of effort and time from my side, as I can't reuse most of the code.

@bassem97 if you insist on introducing this feature I'm happy to review any Pull Request 👍

**Comment by bassem97** [comment]
> Hey all, this formatter is based on a html parser which reorders the attributes. I'm also using TSX a lot and would love react support, but I think this would require a lot of effort and time from my side, as I can't reuse most of the code.
> 
> @bassem97 if you insist on introducing this feature I'm happy to review any Pull Request 👍

@NiklasPor okay i'll see what i can do

**Comment by spacecat** [comment]
Any updates/progress on this feature?

**Comment by Abdrabdra** [comment]
😭😭😭

**Comment by ucimo** [comment]
Still no one interested in this? 

I guess I'll have to try eslint's perfectionist plugin... I really wanted to keep it out of styling concerns. I guess that means trivago's sorting of imports has to go too.

**Comment by spacecat** [comment]
It's been almost a year since I posted my ["thumbs up"](https://github.com/NiklasPor/prettier-plugin-organize-attributes/issues/10#issuecomment-2036433908). 😢

**Comment by NiklasPor** [comment]
> Hey all, this formatter is based on a html parser which reorders the attributes. I'm also using TSX a lot and would love react support, but I think this would require a lot of effort and time from my side, as I can't reuse most of the code.
> 
> [@bassem97](https://github.com/bassem97) if you insist on introducing this feature I'm happy to review any Pull Request 👍

As said before, you basically start from 0 if you implement this for `TSX`, but it's certainly doable. I won't spend any time on this, but feel free to create a PR or even just a separate pretter plugin.

**Comment by spacecat** [comment]
> > Hey all, this formatter is based on a html parser which reorders the attributes. I'm also using TSX a lot and would love react support, but I think this would require a lot of effort and time from my side, as I can't reuse most of the code.
> > 
> > [@bassem97](https://github.com/bassem97) if you insist on introducing this feature I'm happy to review any Pull Request 👍
> 
> As said before, you basically start from 0 if you implement this for `TSX`, but it's certainly doable. I won't spend any time on this, but feel free to create a PR or even just a separate pretter plugin.

Understandable.

Can you at least close the thread, maybe pin it as well, tag it with "need help" and "won't fix" and make an announcement in the README.md so that it gets more attention and hopefully someone can create a PR or a completely new plugin.

Because this plugin gets an insane amount of downloads and this React feature should definitely exist someplace and it would be sad if it gets forgotten.

https://npmtrends.com/prettier-plugin-organize-attributes

**Comment by NiklasPor** [comment]
@spacecat yeah sounds like a good idea. Seems like the packages I created in the past always hunt me in the future – I'm also not actively using this anymore. I sadly don't have the time anymore to start new OSS projects or invest any significant time.

**Comment by Xeonlink** [comment]
I was inspired by this awesome plugin and some issues like react support, svelte support, etc. 

I completely remade this plugin and published named [@xeonlink/prettier-plugin-oragnize-attributes](https://www.npmjs.com/package/@xeonlink/prettier-plugin-organize-attributes). It's support html, angular, vue and also react(jsx, tsx), even svelte. Both plugins has many similar things. testcases, options, etc. Therefore if you just switch prettier-plugin-organize-attribuetes to [@xeonlink/prettier-plugin-oragnize-attributes](https://www.npmjs.com/package/@xeonlink/prettier-plugin-organize-attributes), it will works.

For more details, check out [repository](https://github.com/Xeonlink/prettier-plugin-organize-attributes)

**Comment by tofrankie** [comment]
Looking forward to it 👀
